bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/wrappers/veneer/BigtableHBaseVeneerSettings.java

```diff
@@ -0,0 +1,592 @@
+    return tableAdminSettings;
+  }
+
+  /** Utility to convert {@link Configuration} to {@link BigtableInstanceAdminSettings}. */
+  public BigtableInstanceAdminSettings getInstanceAdminSettings() throws IOException {
+    Preconditions.checkState(
+        isNullOrEmpty(configuration.get(BIGTABLE_EMULATOR_HOST_KEY)),
+        "Instance admin does not support emulator");
+
+    if (instanceAdminSettings == null) {
+      BigtableInstanceAdminSettings.Builder builder =
+          BigtableInstanceAdminSettings.newBuilder().setProjectId(projectId);
+
+      String adminHostOverride = configuration.get(BIGTABLE_ADMIN_HOST_KEY);
+      String endpoint = adminHostOverride + ":" + getPort();
+      LOG.debug("Instance Admin endpoint host:port is %s.", endpoint);
+
+      builder.stubSettings().setHeaderProvider(buildHeaderProvider());
+
+      if (configuration instanceof BigtableExtendedConfiguration
+          || Boolean.parseBoolean(configuration.get(BIGTABLE_USE_SERVICE_ACCOUNTS_KEY))) {
+
+        builder.stubSettings().setCredentialsProvider(buildCredentialProvider());
+      } else if (Boolean.parseBoolean(configuration.get(BIGTABLE_NULL_CREDENTIAL_ENABLE_KEY))) {
+        builder.stubSettings().setCredentialsProvider(NoCredentialsProvider.create());
+      }
+
+      instanceAdminSettings = builder.build();
+    }
+
+    return instanceAdminSettings;
+  }
+  // </editor-fold>
+
+  // <editor-fold desc="Private Helpers">
+  private BigtableDataSettings.Builder buildBigtableDataSettings() throws IOException {
+    BigtableDataSettings.Builder dataBuilder =
+        BigtableDataSettings.newBuilder().setProjectId(projectId).setInstanceId(instanceId);
+
+    String appProfileId = configuration.get(APP_PROFILE_ID_KEY);
+    if (!isNullOrEmpty(appProfileId)) {
+      dataBuilder.setAppProfileId(appProfileId);
+    }
+
+    EnhancedBigtableStubSettings.Builder stubSettings = dataBuilder.stubSettings();
+    String dataHostOverride = configuration.get(BIGTABLE_HOST_KEY);
+    if (!isNullOrEmpty(dataHostOverride)) {
+
+      String port = configuration.get(BIGTABLE_PORT_KEY);
+      if (isNullOrEmpty(port)) {
+        String endpoint = stubSettings.getEndpoint();
+        port = endpoint.substring(endpoint.lastIndexOf(":") + 1);
+      }
+
+      String finalEndpoint = dataHostOverride + ":" + port;
+      LOG.debug("Data API endpoint hostname:portNumber is %s", finalEndpoint);
+
+      stubSettings.setEndpoint(finalEndpoint);
+    }
+
+    stubSettings.setHeaderProvider(buildHeaderProvider());
+
+    if (configuration instanceof BigtableExtendedConfiguration
+        || Boolean.parseBoolean(configuration.get(BIGTABLE_USE_SERVICE_ACCOUNTS_KEY))) {
+      stubSettings.setCredentialsProvider(buildCredentialProvider());
+
+    } else if (Boolean.parseBoolean(configuration.get(BIGTABLE_NULL_CREDENTIAL_ENABLE_KEY))) {
+      stubSettings.setCredentialsProvider(NoCredentialsProvider.create());
+    }
```

Comment:
Please move this to configureCredentials(StubSettings), which will check if an override is necessary and if not it won't touch the credential provider